lib/CodeGen/TargetPassConfig.cpp

```diff
@@ -607,7 +607,8 @@ void TargetPassConfig::addMachinePasses() {
     addPass(&LocalStackSlotAllocationID, false);
   }
 
-  addPass(createPagerandoBinningPass());
+  if (TM->isPagerando())
+    addPass(createPagerandoBinningPass());
```

Comment:
I think it's not ideal to duplicate that for each target. Let's see what LLVM devs prefer.